content/systems/filecoin_blockchain/struct/block/_index.md

```diff
@@ -2,35 +2,67 @@
+The Block structure in the Filecoin blockchain is composed of: i) the Block Header, ii) the list of messages inside the block, and iii) the signed messages. This is represented inside the `FullBlock` abstraction. The messages indicate the required set of changes to apply in order to arrive at a deterministic state of the chain.
 
-{{< embed src="block.id" lang="go" >}}
+The Lotus implementation of the block can be found [here](https://github.com/filecoin-project/lotus/blob/master/chain/types/fullblock.go). It has the following `struct`:
+
+{{<embed src="/externals/lotus/chain/types/fullblock.go"  lang="go" symbol="FullBlock">}}
+
+> **Note:** A block is functionally the same as a block header in the Filecoin protocol. While a block header contains Merkle links to the full system state, messages, and message receipts, a block can be thought of as the full set of this information (not just the Merkle roots, but rather the full data of the state tree, message tree, receipts tree, etc.). Because a full block is large in size, the Filecoin blockchain consists of block headers rather than full blocks. We often use the terms `block` and `block header` interchangeably.
+
+A `BlockHeader` is a canonical representation of a block. BlockHeaders are propagated between miner nodes. From the blockcheader message, a miner has all the required information to apply the associated `FullBlock`'s state and update the chain. In order to be able to do this, the minimum set of information items that need to be included in the `BlockHeader` are shown below and include among others: the miner's address, the Ticket, the [Proof of SpaceTime](post), the CID of the parents where this block evolved from in the IPLD DAG, as well as the messages' own CIDs.
+
+The Lotus implementation of the block header can be found [here](https://github.com/filecoin-project/lotus/blob/master/chain/types/blockheader.go). It has the following `struct`s:
+
+{{<embed src="/externals/lotus/chain/types/blockheader.go"  lang="go" symbol="BlockHeader">}}
+
+{{<embed src="/externals/lotus/chain/types/blockheader.go"  lang="go" symbol="Ticket">}}
+
+{{<embed src="/externals/lotus/chain/types/electionproof.go"  lang="go" symbol="ElectionProof">}}
+
+{{<embed src="/externals/lotus/chain/types/blockheader.go"  lang="go" symbol="BeaconEntry">}}
+
+The `BlockHeader` structure has to refer to the TicketWinner of the current round which ensures the correct winner is passed to [ChainSync](chainsync).
+
+```go
+func IsTicketWinner(vrfTicket []byte, mypow BigInt, totpow BigInt) bool
+```
+
+The `Message` structure has to include the source (`From`) and destination (`To`) addresses, a `Nonce` and the `GasPrice`.
+
+The Lotus implementation of the message can be found [here](https://github.com/filecoin-project/lotus/blob/master/chain/types/message.go). It has the following structure:
```

Comment:
all embed from github have source permanent links no need to duplicate here. check the suggested change below.